packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+  Box,
+  Button,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuLabel,
+  DropdownMenuTrigger,
+  Icon,
+  Text,
+} from "@prismicio/editor-ui";
+import { useReducer } from "react";
+
+import { useOnboardingProgress } from "@/features/onboarding/helpers";
+import { OnboardingStepDialog } from "@/features/onboarding/OnboardingStepDialog";
+import type { OnboardingStep } from "@/features/onboarding/types";
+
+type DialogState = {
+  isOpen: boolean;
+  step: OnboardingStep;
+};
+
+export const OnboardingProgressStepper = () => {
+  const { completedStepCount, steps, isStepComplete } = useOnboardingProgress();
+
+  const [dialog, setDialog] = useReducer(
```

Comment:
Yeah that's fair, I'll change to two useStates, as with only one I would have to spread the current value on some calls, and IMO that's doesn't look great

Reply:
But why two?

Reply:
Since the radix dialog uses an `open` prop to have a open and closing animation, I have one for that, and another for the `step` to pass to the dialog. I avoid setting the step to `undefined` to allow the closing animation to happen.

Reply:
yeah, i meant that before (where you were using useReducer) you had the state in one object, but you split it into two states when you moved to useState. But that's ok, both work 👍

Reply:
Yeah I used two to avoid doing stuff like `setDialog((prev) => ({ ...prev: open: false }))` just to open the dialog, this way you only change the `setDialogOpen` in that case.